staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -55,16 +56,18 @@ import (
 )
 
 const (
-	aesCBCTransformerPrefixV1    = "k8s:enc:aescbc:v1:"
-	aesGCMTransformerPrefixV1    = "k8s:enc:aesgcm:v1:"
-	secretboxTransformerPrefixV1 = "k8s:enc:secretbox:v1:"
-	kmsTransformerPrefixV1       = "k8s:enc:kms:v1:"
-	kmsTransformerPrefixV2       = "k8s:enc:kms:v2:"
-	kmsPluginHealthzInterval     = 1 * time.Minute
-	kmsPluginHealthzNegativeTTL  = 3 * time.Second
-	kmsPluginHealthzPositiveTTL  = 20 * time.Second
-	kmsAPIVersionV1              = "v1"
-	kmsAPIVersionV2              = "v2"
+	aesCBCTransformerPrefixV1        = "k8s:enc:aescbc:v1:"
+	aesGCMTransformerPrefixV1        = "k8s:enc:aesgcm:v1:"
+	secretboxTransformerPrefixV1     = "k8s:enc:secretbox:v1:"
+	kmsTransformerPrefixV1           = "k8s:enc:kms:v1:"
+	kmsTransformerPrefixV2           = "k8s:enc:kms:v2:"
+	kmsPluginHealthzPositiveInterval = 1 * time.Minute
+	kmsPluginHealthzNegativeInterval = 10 * time.Second
+	kmsPluginDEKReuseInterval        = 3 * time.Minute
```

Comment:
Important.

Reply:
What's stopping us from making this 24 hours or more (or removing it altogether)?

Also something like "writeDEKMaxAge" might be a clearer name.

Reply:
The gist is that it makes it impossible to have a "passive" storage migration.  With the current approach, you update the key ID your plugin is using and then can simply schedule a migration for `10+N` minutes later where `N` is how long it takes your plugin to pick up new config.  At that point, you are guaranteed to either migrate to the new key or get errors during the migration.  If the API server coasted forever on the last DEK, you would need to actively check if it had observed the new key ID before starting a migration - otherwise it could keep using the old DEK and your storage migration would not do what you thought it did.  I think we can iterate on making this approach better but for now I think we need to stick with the safe approach.